tests/printer_test.py

```diff
@@ -608,3 +608,28 @@ def test_parse_generic_format_attr():
     printer = Printer(stream=file, print_generic_format=True)
     printer.print_op(module)
     assert file.getvalue().strip() == expected.strip()
+
+
+def test_parse_dense_mlir():
+    """
+    Test that we can parse attributes using generic formats.
+    """
+    prog = """
+    %0 = "arith.constant"() {"value" = dense<[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]> : tensor<2x3xf64>} : () -> tensor<2x3xf64>
```

Comment:
Ideally we also need a test to cover the case where I<1 for i in shape.
So a test with `dense<[[1.0, 2.0, 3.0], [4.0, **0.0,** 6.0]]>` could be useful.

Reply:
It's not on the shape here, but rather on the values.
I think we should add this, since we could have problems where `0.0` is put in an `if ...`.
Though george, I have to say that this is exactly why I think we should explicitely write `if .. is not None`, so we don't have this kind of issues in general.